src/app.ts

```diff
@@ -92,11 +93,23 @@ app.use(requestContext.middleware());
 if (process.env.NODE_ENV !== "production")
   app.use("/voyager", voyagerMiddleware({ endpointUrl: "/graphql" }));
 
-app.get("/", (req, res) =>
-  res.json({
-    "talawa-version": "v1",
-    status: "healthy",
-  })
-);
+const redisClient = new Redis();
+app.get("/", (req, res) => {
+  // Check if Redis is connected before responding
+
+  if (redisClient.status == "ready") {
+    res.json({
+      "talawa-version": "v1",
+      status: "healthy",
+      redisStatus: "connected",
+    });
+  } else {
+    res.json({
+      "talawa-version": "v1",
+      status: "unhealthy",
+      redisStatus: "disconnected",
+    });
+  }
+});
```

Comment:
This is not the correct place to make this change. If the Redis server fails to start, the Express + GraphQL server should crash. The `/` endpoint is just a safety endpoint that the users can ping for a sanity check.

Reply:
@EshaanAgg 
Yes, the else part of the code is unnecessary since the express+Graphql server crashes when Redis server fails to start. Will remove it. Thanks.

Reply:
@EshaanAgg  But the issue mentioned to specify the connection status at http://localhost:4000/, so  the code  indicating the successful connection status should be present right?

Reply:
The issue is moot then. Please make sure that the service should fail with appropriate error message if Redis is not available, and the index homepage should always return the same message.

This message is also used in the Health Check workflow so please make sure that the object returned is exactly the same in case of any modifications.

Reply:
@EshaanAgg 
I made some changes which resulted in this response when the redis-server is not connected.
![Screenshot 2024-01-02 174624](https://github.com/PalisadoesFoundation/talawa-api/assets/135726978/2bea7c94-a240-44f7-86c9-502d86bc5a85)

Reply:
But the issue is when the redis is not connected error messages appear on the console and after MaxRetriesPerRequest 
 http://localhost:4000/ fails to respond the status. In my opinion on handling this error the issue can be resolved.
Any recommendations/suggestions regarding this?
![Screenshot 2024-01-02 184147](https://github.com/PalisadoesFoundation/talawa-api/assets/135726978/69df6a3a-7966-418d-9943-6371b947055b)

Reply:
There are two options to proceed:
1. If the resolvers can function without Redis, then we should have two possible messages on the home page as you have been doing. The error of not being able to connect to Redis should be logged properly in console.

2. If the resolvers and the rest of app can't function without Redis, then we should delay starting the server until the connection is made, and should kill the server if the same fails. Use the [clientReconnect](https://github.com/redis/node-redis/blob/HEAD/docs/client-configuration.md#reconnect-strategy) option to fix this and define a custom strategy that kills the server if connection fails.

Reply:
Please research what the current implementation is and implement the required fix.

Reply:
Sure, will fix this soon.